pos.lua

```diff
@@ -181,41 +179,67 @@ function areas:getPos(playerName)
 	return areas:sortPos(pos1, pos2)
 end
 
-function areas:setPos1(playerName, pos)
-	areas.pos1[playerName] = posLimit(pos)
-	areas.markPos1(playerName)
-end
+function areas:setPos1(name, pos)
+	local old_pos = areas.pos1[name]
+	pos = posLimit(pos)
+	areas.pos1[name] = pos
 
-function areas:setPos2(playerName, pos)
-	areas.pos2[playerName] = posLimit(pos)
-	areas.markPos2(playerName)
+	local entity = minetest.add_entity(pos, "areas:pos1")
+	if entity then
+		local luaentity = entity:get_luaentity()
+		if luaentity then
+			luaentity.player = name
+		end
+	end
+
+	if old_pos then
+		for object in core.objects_inside_radius(old_pos, 0) do
```

Comment:
I'd feel safer by having a small tolerance to compensate for any float -> double -> float conversions:
```suggestion
		for object in core.objects_inside_radius(old_pos, 0.01) do
```

Same comment for `areas:setPos2`.